shared_model/schema/block.proto

```diff
@@ -12,10 +12,16 @@ message Block {
   // everything that should be signed:
   message Payload {
     repeated Transaction transactions = 1;
-    uint32 tx_number = 2;  // the number of transactions inside. Maximum 16384 or 2^14
-    uint64 height = 3;          // the current block number in a ledger
-    bytes prev_block_hash = 5;  // Previous block hash
+    uint32 tx_number = 2;  ///< The number of accepted transactions inside.
+                           ///< Maximum 16384 or 2^14.
+    uint64 height = 3;          ///< The current block number in a ledger.
+    bytes prev_block_hash = 5;  ///< Previous block hash.
     uint64 created_time = 6;
+
+    /// Hashes of the transactions that did not pass stateful validation.
+    /// Needed here to be able to guarantee the client that this transaction
+    /// was not and will never be executed.
+    repeated bytes rejected_transactions_hashes = 7;
```

Comment:
can we make this field optional? check out batch meta in transaction.proto. That way genesis block will still work, otherwise please check that genesis block in example folder if it launches with iroha version in your branch

Reply:
I think this field already has the ability of optional value to contain nothing (as an empty sequence). The functions regarding genesis block seem to be unchanged, as the new block should deserialize properly without the new field (please see test `[irohad_test](https://github.com/hyperledger/iroha/blob/feature/rejected-txs-in-block/test/system/irohad_test.cpp)`).